model/common/src/icon4py/model/common/grid/vertical.py

```diff
@@ -630,48 +627,43 @@ def _decay_func(
 def _check_and_correct_layer_thickness(
     vertical_coordinate: data_alloc.NDArray,
     vct_a: data_alloc.NDArray,
-    vertical_config: VerticalGridConfig,
-    grid: base.BaseGrid,
+    SLEVE_minimum_layer_thickness_1: ta.wpfloat,
+    SLEVE_minimum_relative_layer_thickness_1: ta.wpfloat,
+    SLEVE_minimum_layer_thickness_2: ta.wpfloat,
+    SLEVE_minimum_relative_layer_thickness_2: ta.wpfloat,
+    lowest_layer_thickness: ta.wpfloat,
     array_ns: ModuleType = np,
 ) -> data_alloc.NDArray:
-    num_levels = vertical_config.num_levels
-    num_cells = grid.num_cells
-    ktop_thicklimit = array_ns.asarray(num_cells * [num_levels], dtype=float)
+    num_cells = vertical_coordinate.shape[0]
+    num_levels = vct_a.shape[0] - 1
```

Comment:
The `vertical_coordinate` is a (nume_cell, nlev + 1) field right? I think it is a bit more consistent if you used the same field for the sizes here
```suggestion
    num_cells = vertical_coordinate.shape[0]
    num_levels = vertical_coordinate.shape[1] -1
```